src/models/cases.js

```diff
@@ -1,35 +1,83 @@
 'use strict'
 
-module.exports = CasesFactory
+module.exports = {
+  CasesFactory,
+  convertFromExternal,
+  convertToExternal
+}
 
-const CaseFactory = require('./db/case')
-const Log = require('../lib/log')
 const config = require('../lib/config')
 const NotificationWorker = require('../lib/notificationWorker')
 const CaseExpiryMonitor = require('../lib/caseExpiryMonitor')
+const db = require('./db/case')
+const getCaseByUuid = require('./db/case').getCaseByUuid
 const cc = require('five-bells-condition')
+const uri = require('../lib/uri')
+const moment = require('moment')
 const UnprocessableEntityError = require('five-bells-shared').UnprocessableEntityError
 const UnmetConditionError = require('five-bells-shared').UnmetConditionError
 const NotFoundError = require('five-bells-shared').NotFoundError
-const db = require('./db/cases')
+const InvalidBodyError = require('five-bells-shared/errors/invalid-body-error')
+const validator = require('../lib/validator')
+const _ = require('lodash')
 
-CasesFactory.constitute = [CaseFactory, Log, NotificationWorker, CaseExpiryMonitor]
-function CasesFactory (Case, log, notificationWorker, caseExpiryMonitor) {
+function convertFromExternal (data) {
+  // ID is optional on the incoming side
+  const caseObject = {
+    execution_condition: data.execution_condition,
+    expires_at: new Date(data.expires_at),
+    notaries: data.notaries,
+    notification_targets: data.notification_targets,
+    state: data.state
+  }
+  if (data.id) {
+    caseObject.id = uri.parse(data.id, 'case').id.toLowerCase()
+  }
+  data = caseObject
+  return data
+}
+
+function convertToExternal (data) {
+  const externalCase = {
+    id: uri.make('case', data.id.toLowerCase()),
+    execution_condition: data.execution_condition,
+    expires_at: moment(data.expires_at).toISOString(), // format('YYYY-MM-DD HH:mm:ss');
+    notaries: data.notaries,
+    notification_targets: data.notification_targets,
+    state: data.state,
+    exec_cond_fulfillment: data.exec_cond_fulfillment
+  }
+  data = externalCase
+  return data
+}
+
+function isFinalized (caseObject) {
+  return caseObject.state === 'executed' || caseObject.state === 'rejected'
+}
+
+CasesFactory.constitute = [NotificationWorker, CaseExpiryMonitor]
+function CasesFactory (notificationWorker, caseExpiryMonitor) {
```

Comment:
@clark800 it is possible to get rid of the class and make this purely functional, but I'm getting some weird `constitute` dependency issues. I'd prefer to address this in a separate PR, since it's not central to the task of refactoring the db.